### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by cococococococo** [brocoders/nestjs-boilerplate#1435] [General]
Hello
I am trying to use the boilerplate and to understand its functioning (and I am a bit new to nestjs, so I beg your pardon if my questions seems silly).
Question: 
1./ What is the use of folder /domain in folders and why do we need them ? My understanding is this is to define the class of the objet which what the module is representing, but should not it be same than the .entity file?
2./ Why don't the seeder are not located inside the folders of the module ?
3./ Is there any way to generate the migration file based on the .entity file ? 
4./ I have tried to activite the synchronize but it seems it doesn't work
5./ I have a bit some difficulties to understand how you have made your i18n structure. Normally we should have something like this (one json per language) :
src
└── i18n
    ├── en
    │   └── app_language.json
    └── nl
        └── app_language.json
        
        
But in the boilerplate this is different it is like you have separated in many json. 
How to use  ? Why not everything in a single json file?
6./ I have difficulties to understand what is persistence and why we need this.

Thanks in advance for your feedback.
7./ What is the use of mappers in the user/infrastructure/relational ?

**Comment by Shchepotin** [comment]
@cococococococo 

1,6,7. We need all related stuff for Hexagonal Architecture. More info here: https://github.com/brocoders/nestjs-boilerplate/blob/main/docs/architecture.md
2. We decided to use a database module for seeder and migrations because this approach is common.
3. https://github.com/brocoders/nestjs-boilerplate/blob/main/docs/database.md#generate-migration
4. We don't recommend using this option, but it should be supported by TypeORM, if you have some trouble with it, please create the issue in their repo.
5. We have the structure that you described but use separate files. You can use one file for all translations.
